test/pylib/resource_gather.py

```diff
@@ -89,23 +89,22 @@ def  run_process(self, args: list[str], timeout: float, env: dict = None) -> tup
             env = os.environ.copy()
         p = subprocess.Popen(
             args,
-            stdout=subprocess.PIPE,
+            stdout=output_file.open('w'),
```

Comment:
Does the subprocess.Popen automatically close the file?

Reply:
Yes, `Popen` have an argument `close_fds=True`.

Reply:
I would specify it here explicitly, even though it is default, I would like to not depend on default values when it comes to hanging file descriptors